File: validSequence.py
# O(n) time | O(1) space

def validSequence(array, sequence):
    arrIdx = 0
    seqIdx = 0
    while arrIdx < len(array) and seqIdx < len(sequence):
        if array[arrIdx] == sequence[seqIdx]:
            seqIdx += 1
        arrIdx += 1
    return seqIdx == len(sequence)


if __name__ == "__main__":
    print(validSequence(array=[5, 3, 8, 11, -7,
          2, -1, 10, 7], sequence=[3, 2, 10, 7]))

File: validSequence.cpp
#include <iostream>
using namespace std;
#include <vector>


bool validSequence(vector<int> array, vector<int> sequence){
    int arrayIdx = 0;
    int seqIdx = 0;
    while (arrayIdx < array.size() && seqIdx < sequence.size()){
        if (array[arrayIdx] == sequence[seqIdx]){
            seqIdx += 1;
            

        }
        arrayIdx += 1;
    }
    return seqIdx == sequence.size();

}

int main(){
    vector<int> array {3, 5, 1, 22, 25, 4, 8, -1, 16};
    vector<int> sequence {1, 25, 8, -1 };
    bool results = validSequence(array, sequence);
    cout << results;

}